src/System.IO.Pipes/src/System/IO/Pipes/PipeStream.Windows.cs

```diff
@@ -412,6 +412,32 @@ internal static unsafe Interop.Kernel32.SECURITY_ATTRIBUTES GetSecAttrs(HandleIn
             return secAttrs;
         }
 
+        internal static unsafe Interop.Kernel32.SECURITY_ATTRIBUTES GetSecAttrs(HandleInheritability inheritability, PipeSecurity pipeSecurity, ref GCHandle pinningHandle)
+        {
+            Interop.Kernel32.SECURITY_ATTRIBUTES secAttrs = default(Interop.Kernel32.SECURITY_ATTRIBUTES);
+            if (pipeSecurity != null || (inheritability & HandleInheritability.Inheritable) != 0)
+            {
+                secAttrs = new Interop.Kernel32.SECURITY_ATTRIBUTES();
+                secAttrs.nLength = (uint)sizeof(Interop.Kernel32.SECURITY_ATTRIBUTES);
+
+                if ((inheritability & HandleInheritability.Inheritable) != 0)
+                {
+                    secAttrs.bInheritHandle = Interop.BOOL.TRUE;
+                }
+
+                byte[] securityDescriptor = pipeSecurity.GetSecurityDescriptorBinaryForm();
```

Comment:
Since it is an "or" on the if statement you can hit this line if `pipeSecurity` is null.

Reply:
hrgg forgot to add that if in the refactor. Thanks for catching it.